ee/app/settings/server/settings.ts

```diff
@@ -54,9 +56,18 @@ SettingsEvents.on('change-setting', (record: ISettingRecord, value: ISettingNoti
 	}
 });
 
-
-onValidateLicenses(() => {
+function updateSettings(): void {
 	const enterpriseSettings = SettingsModel.findEnterpriseSettings();
 
 	enterpriseSettings.forEach((record: ISettingRecord) => settings.storeSettingValue(record, false));
+}
+
+
+Meteor.startup(() => {
+	updateSettings();
+
+	// If there was no license loaded, add a callback to update the settings once one is added
+	if (!isEnterprise()) {
```

Comment:
It should always run callback ance the license changes, even when became invalid